controllers/kubevirtmachine_controller.go

```diff
@@ -154,16 +157,19 @@ func (r *KubevirtMachineReconciler) Reconcile(goctx gocontext.Context, req ctrl.
 	}
 
 	// Handle non-deleted machines
-	return r.reconcileNormal(machineContext)
+	if result, err := r.reconcileNormal(machineContext); err != nil || !machineContext.KubevirtMachine.Status.Ready {
+		return result, err
+	}
+
+	// Update the providerID on the Node
+	// The ProviderID on the Node and the providerID on  the KubevirtMachine are used to set the NodeRef
+	// This code is needed here as long as there is no Kubevirt cloud provider setting the providerID in the node
+	return r.updateNodeProviderID(machineContext)
```

Comment:
minor nit here.

It's slightly concerning that the `result` from reconcileNormal could be overlooked in the future. I get that the way this is written this ensures that the `reconcileNormal()` result will not try to re-enqueue anything after `Status.Ready == true`. It doesn't feel right to ignore the `result` from reconcileNormal() though.

Wouldn't something like what I have below logically be the same thing as what you have now, but not run the risk of ignoring the reconcileNormal's `result` at some point in the future?

```
if !machineContext.KubevirtMachine.Status.Ready {
    return r.reconcileNormal(machineContext)
} else {
    return r.updateNodeProviderID(machineContext)
}
```

Reply:
good suggestion, to be safe in the future

Reply:
Changed, please review